src/main/kotlin/blackjack/domain/blackjack/BlackJackResult.kt

```diff
@@ -1,23 +1,34 @@
 package blackjack.domain.blackjack
 
 import blackjack.domain.card.Cards
+import blackjack.domain.common.Money
 import blackjack.domain.player.Dealer
 import blackjack.domain.player.Participant
 import blackjack.domain.player.Player
-import blackjack.domain.score.Match
+import blackjack.domain.player.Players
 
 data class BlackJackResult(
-    val playerResults: List<PlayerResult>,
-    val dealerResult: DealerResult
+    private val dealer: Dealer,
+    private val players: Players
 ) {
-    companion object {
-        fun of(players: List<Player>, dealer: Dealer): BlackJackResult {
-            val matchByPlayer = players.associateWith { !dealer.match(it) }
-            val playerResults = matchByPlayer.entries
-                .map { PlayerResult.of(it.key, it.value) }
-            val dealerResult = DealerResult.of(dealer, matchByPlayer.values.map { !it }.toList())
-            return BlackJackResult(playerResults, dealerResult)
-        }
+
+    fun dealerResult(): ParticipantResult {
+        return ParticipantResult.of(dealer)
+    }
+
+    fun playersResult(): List<ParticipantResult> {
+        return players.players.map { ParticipantResult.of(it) }
+    }
+
+    fun dealerProfit(): ParticipantProfitResult {
+        val profitByPlayer: Map<Player, Money> = players.profit(dealer)
```

Comment:
```suggestion
        val profitByPlayer = players.profit(dealer)
```

타입에 대해 생략해줘도 될 것 같아요 : )